cca_realm_claims_test.go

```diff
@@ -0,0 +1,273 @@
+
+	err = c.SetExtensibleMeasurements([][]byte{})
+	expectedErr = "missing mandatory claim cca-realm-extended-measurements:"
+	assert.EqualError(t, err, expectedErr)
+
+	err = c.SetHashAlgID("blank")
+	expectedErr = "wrong syntax for claim: wrong syntax"
+	assert.EqualError(t, err, expectedErr)
+
+	err = c.SetPubKey([]byte("random-key"))
+	expectedErr = "wrong syntax for claim: length 10 (cca-realm-public-key MUST be 97 bytes)"
+	assert.EqualError(t, err, expectedErr)
+
+	err = c.SetPubKeyHashAlgID("")
+	expectedErr = "invalid null string set for cca-realm-pubkey-hash-alg-id"
+	assert.EqualError(t, err, expectedErr)
+}
+
+func Test_Cca_Realm_Claims_ToCBOR_invalid(t *testing.T) {
+	c, err := NewClaims()
+	require.NoError(t, err)
+
+	expectedErr := `validation of CCA realm claims failed: validating cca-realm-challege claim : missing mandatory claim`
+
+	_, err = c.ToCBOR()
+
+	assert.EqualError(t, err, expectedErr)
+}
+
+func Test_Cca_Realm_Claims_ToCBOR_all_claims(t *testing.T) {
+	c := mustBuildValidCcaRealmClaims(t)
+
+	expected := mustHexDecode(t, testEncodedCcaRealmClaimsAll)
+
+	actual, err := c.ToCBOR()
+
+	assert.NoError(t, err)
+	assert.Equal(t, expected, actual)
+}
+
+func Test_Cca_Realm_Claims_FromCBOR_ok(t *testing.T) {
+	buf := mustHexDecode(t, testEncodedCcaRealmClaimsAll)
+
+	var c CcaRealmClaims
+	err := c.FromCBOR(buf)
+	assert.NoError(t, err)
+
+	// mandatory
+	expectedChallenge := testChallenge
+	actualChallenge, err := c.GetChallenge()
+	assert.NoError(t, err)
+	assert.Equal(t, expectedChallenge, actualChallenge)
+
+	expectedPersonalizationVal := testPersonalizationVal
+	actualPersonalizationVal, err := c.GetPersonalizationValue()
+	assert.NoError(t, err)
+	assert.Equal(t, expectedPersonalizationVal, actualPersonalizationVal)
+
+	expectedInitMeas := testInitMeas
+	actualInitMeas, err := c.GetInitialMeasurements()
+	assert.NoError(t, err)
+	assert.Equal(t, expectedInitMeas, actualInitMeas)
+
+	expectedExtensibleMeas := testExtensibleMeas
+	actualExtensibleMeas, err := c.GetExtensibleMeasurements()
+	assert.NoError(t, err)
+	assert.Equal(t, expectedExtensibleMeas, actualExtensibleMeas)
+
+	expectedHashAlgID := testHashAlgID
+	actualHashAlgID, err := c.GetHashAlgID()
+	assert.NoError(t, err)
+	assert.Equal(t, expectedHashAlgID, actualHashAlgID)
+
+	expectedPubKey := testPubKey
+	actualPubKey, err := c.GetPubKey()
+	assert.NoError(t, err)
+	assert.Equal(t, expectedPubKey, actualPubKey)
+
+}
+
+func Test_Cca_Realm_Claims_FromCBOR_bad_input(t *testing.T) {
+	buf := mustHexDecode(t, testNotCBOR)
+
+	expectedErr := "CBOR decoding of CCA realm claims failed: unexpected EOF"
+
+	var c CcaRealmClaims
+	err := c.FromCBOR(buf)
+
+	assert.EqualError(t, err, expectedErr)
+}
+
+func Test_Cca_Realm_Claims_FromCBOR_missing_mandatory_claims(t *testing.T) {
+	buf := mustHexDecode(t, testEncodedCcaRealmClaimsMissingMandNonce)
+
+	expectedErr := "validation of CCA realm claims failed: validating cca-realm-challege claim : missing mandatory claim"
```

Comment:
spurious whitespace (here and below)

```suggestion
	expectedErr := "validation of CCA realm claims failed: validating cca-realm-challege claim: missing mandatory claim"
```